### ISSUE TITLE
Some improvements ideas.

### ISSUE THREAD

**Issue by janoamaral** [paulirish/lite-youtube-embed#48]
I've working on a similar implementation. Here is a couple of ideas.
- Used the a tag `<a href="https://youtu.be/3vBwRfQbXkg" class="ytlight">HOW TO FUNK IN TWO MINUTES</a>`. If the user has JavaScript disabled, it can go to the video anyway.
- Lazy load the thumbnail.
- The quality of the thumbnail to load is decided with `NetworkInformation.effectiveType`.

[POC](https://codepen.io/logico-ar/pen/ZEYPrwG)

[Test page lite-youtube-embed](https://lab.logico.ar/ytlight/lite-youtube-embed/) 
[Test page ytlight](https://lab.logico.ar/ytlight/ytlight/)

**Comment by simevidas** [comment]
Isn’t adding an `<a>` element the responsibility of the website that uses the component, not the component itself?

**Comment by paulirish** [comment]
- since the first release, we've featured the Progressive Enhancement style usage: https://github.com/paulirish/lite-youtube-embed/#pro-usage-load-w-js-deferred-aka-progressive-enhancement. So yeah.. same deal.
- lazy loading is covered in #57 
- I just added support for loading a better quality thumbnail in #167 . i wouldn't trust effectiveType with much, personally. But.. the motivation is a good one. :)
